packages/ui-contexts/src/LoginContext.ts

```diff
@@ -0,0 +1,29 @@
+import { createContext } from 'react';
+
+export type LoginService = {
+	clientConfig: unknown;
+
+	title: string;
+	service: string;
+
+	buttonLabelText?: string;
+	buttonLabelColor?: string;
+	buttonColor?: string;
+};
+
+export type LoginContextValue = {
+	loginWithPassword: (user: string | object, password: string) => Promise<void>;
+	loginWithToken: (user: string) => Promise<void>;
+	logout: () => Promise<void>;
+
+	queryAllServices(): [subscribe: (onStoreChange: () => void) => () => void, getSnapshot: () => LoginService[]];
+	loginWithService<T extends LoginService>(service: T): () => Promise<true>;
+};
+
+export const LoginContext = createContext<LoginContextValue>({
```

Comment:
I would merge `LoginContext` and `UserContext` into one (e.g. `AuthenticationContext`) as their APIs so correlated.